src/api/cvc4cpp.cpp

```diff
@@ -589,6 +589,17 @@ const static std::unordered_set<Kind, KindHashFunction> s_indexed_kinds(
      FLOATINGPOINT_TO_FP_UNSIGNED_BITVECTOR,
      FLOATINGPOINT_TO_FP_GENERIC});
 
+/** Set of kinds where API term structure differs from internal structure
+ *  The API takes a "higher-order" perspective and treats functions as well
+ *  as datatype constructors/selectors/testers as terms
+ *  but interally they are not
+ */
+const static std::unordered_set<CVC4::Kind, CVC4::kind::KindHashFunction>
```

Comment:
I'd prefer if this were a utility function not data.

Reply:
Cool, just made that change with ee9583187c5e420b3b20bd6a9a6bef0fd338bc01. I also made it so that all the special casing uses this new utility function. Should be cleaner that way.